### PR TITLE
chore(config): mark `cssVarShim` as deprecated

### PR DESCRIPTION
**Pull request by alicewriteswrongs**
This marks the `extras.cssVarsShim` field on the Stencil configuration interface as deprecated, and introduces TODO comments for removing all of the relevant code (that I could find).

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Build (`npm run build`) was run locally and any changes were pushed
- [ ] Unit tests (`npm test`) were run locally and passed
- [ ] E2E Tests (`npm run test.karma.prod`) were run locally and passed
- [ ] Prettier (`npm run prettier`) was run locally and passed

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

We want to remove Stencil's built-in support for including polyfills for old browsers. Before we remove it, we want to mark fields for deprecation.

## What is the new behavior?

the field is renamed from `cssVarShim` to `__deprecated_cssVarShim`.

## Does this introduce a breaking change?

- [x] Yes
- [ ] No

### CODE REVIEW

**Comment by alicewriteswrongs** [comment]
note: I updated this branch to include the `@deprecated` tag in the JSDoc for the field and updated the commit message to include the `BREAKING CHANGES:` bit.

**Review by tanner-reits — APPROVED** [review]

**Review by rwaskiewicz — CHANGES_REQUESTED** [review]
Had one small `TODO` addition ask. 

Before we move forward w this, can you update `BREAKING_CHANGES.md` and link the PRs for the doc site back to this PR?

File: src/client/client-patch-browser.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@ export const patchBrowser = (): Promise<d.CustomElementsDefineOptions> => {
     consoleDevInfo('Running in development mode.');
   }
 
+  // TODO(STENCIL-659): Remove code implementing the CSS variable shim
   if (BUILD.cssVarShim) {
     // shim css vars
     plt.$cssShim$ = (win as any).__cssshim;
```
[the remark this answers is not in the document]
Comment (alicewriteswrongs) [reply src/client/client-patch-browser.ts:16]: Sure, I'll call it out. Worst case, when we circle around to actually delete the code we decide some parts have to be left in for some reason and delete / change some comments

**Comment by rwaskiewicz** [comment]
Docs: https://github.com/ionic-team/stencil-site/pull/950

**Review by rwaskiewicz — APPROVED** [review]
LGTM, although we may want to propagate the changes from https://github.com/ionic-team/stencil-site/pull/950 to here

**Comment by alicewriteswrongs** [comment]
agreed, just pushed that change!

**Pull request merged**